depthai_ros_driver/include/depthai_ros_driver/device_ros.hpp

```diff
@@ -188,7 +188,7 @@ class DeviceROS : public dai::DeviceBase {
                 msg.data = reader.bufferData();
 
                 // publish data
-                pub.publish(adapt_dai2ros<MsgType>::convert(msg, name));
+                publish(pub, msg, name);
```

Comment:
based on conversation, it need std::move(msg) here?

Reply:
Maybe we can just use your idea of non-const reference.